extensions/amp-story/1.0/media-pool.js

```diff
@@ -545,7 +551,7 @@ export class MediaPool {
       return;
     }
 
-    componentEl.getImpl().then((impl) => {
+    return componentEl.getImpl().then((impl) => {
```

Comment:
Make `maybeResetAmpMedia_` async so the `swap..MediaIntoDom_` can wait for this.

Reply:
It's pretty cool that `Promise.all` handles `undefined` so well but let's change this method to be `@return {!Promise}` it'll make things easier